### ISSUE TITLE
Improve API to filter `TsGroup`

### ISSUE THREAD

**Issue by alejoe91** [pynapple-org/pynapple#243]
Hi @gviejo 

As mentioned previously, this is to suggest an easier API to filter `TsGroup`. Specifically, this is how filtering the units table from the `pynwb.Units` object looks like:

```
units = nwbfile.units[:]
# units is now a pandas dataframe
pass_qc = units[(units.isi_violations_ratio < 0.5) & 
                (units.firing_rate > 0.1) &
                (units.presence_ratio > 0.95) &
                (units.device_name == 'ProbeB')]
# or
pass_qc = units.query("isi_violations_ratio < 0.5 and ...")
```

The same operation in pynapple is much mroe convoluted:
```
spikes = data['units']

tsgroup = (spikes.getby_category('device_name')['ProbeB']
           .getby_threshold('isi_violations_ratio', 0.5, op='<')
           .getby_threshold('presence_ratio', 0.95, op='>')
           .getby_threshold('firing_rate',0.1,op='>'))
```

Would it be possible to simplify such queries with a simil-pandas API?

**Comment by BalzaniEdoardo** [comment]
Hi Ale! Thanks for your suggestion. I have implemented a new `__getattr__` and a new `__getitem__` and `__setitem__` logic for the `TsGroup`. The PR is currently under revision, if you want to take a look. I'll ping you when it is merged!

Cheers
Edo

**Comment by alejoe91** [comment]
> Hi Ale! Thanks for your suggestion. I have implemented a new `__getattr__` and a new `__getitem__` and `__setitem__` logic for the `TsGroup`. The PR is currently under revision, if you want to take a look. I'll ping you when it is merged!
> 
> Cheers Edo

Thannks Edo! I'll take a look and give it a try :)

**Comment by gviejo** [comment]
Solved in #263
